### PR TITLE
accept not found errors during cleanup of openstack resources

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
accept not found errors during cleanup of openstack resources

**Release note**:
```release-note cloud provider
Openstack: made cluster resource cleanup idempotent
```

### CODE REVIEW

File: api/pkg/provider/cloud/openstack/provider.go [hunk 1]
```diff
@@ -135,17 +135,22 @@ func (os *Provider) CleanUpCloudProvider(cloud *kubermaticv1.CloudSpec) error {
 	}
 
 	if cloud.Openstack.NetworkCreated {
-
 		if _, err = detachSubnetFromRouter(netClient, cloud.Openstack.SubnetID, cloud.Openstack.RouterID); err != nil {
-			return fmt.Errorf("failed to detach subnet from router: %v", err)
+			if err.Error() != "Resource not found" {
```
Comment (guusvw) [inline api/pkg/provider/cloud/openstack/provider.go]: Could we used a typed error? 
I feel really bad comparing the error message.

**Review by guusvw — CHANGES_REQUESTED** [review]

File: api/pkg/provider/cloud/openstack/provider.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/provider/cloud/openstack/provider.go]: Sure!

Comment (guusvw) [reply api/pkg/provider/cloud/openstack/provider.go]: Or at least define the error message in a variable/constant

**Review by guusvw — APPROVED** [review]
LGTM

**Pull request merged**